author/topics/automation/json_to_text.adoc

```diff
@@ -0,0 +1,581 @@
+* `{{forloop.index0}}` gives the zero-based position of item `item` in the parent array `arr`.
+
+Will render as:
+[source,text]
+----
+=== 0 lorem
+
+This section is about lorem.
+
+=== 1 ipsum
+
+This section is about ipsum.
+
+=== 2 dolor
+
+This section is about dolor.
+----
+
+--
+
+
+
+=== Accessing objects
+
+
+==== Size
+
+Similar to arrays, the number of key-value pairs within an object can be
+obtained by `{{objectname.size}}`.
+
+EXAMPLE:
+--
+Given:
+
+object.json
+[source,json]
+----
+---
+name: Lorem ipsum
+desc: dolor sit amet
+----
+
+And the block:
+[source,asciidoc]
+------
+[json2text,object.json,data]
+----
+=== {{data.name}}
+
+{{data.desc}}
+----
+------
+
+The file path is `object.json`, and context name is `data`.
+`{{data.size}}` evaluates to the size of the object.
+
+Will render as:
+[source,asciidoc]
+----
+=== Lorem ipsum
+
+dolor sit amet
+----
+
+--
+
+==== Enumeration and context
+
+The following syntax is used to enumerate key-value pairs within an object:
+
+[source,asciidoc]
+--
+{% for item in object_name %}
+  {{item[0]}}, {{item[1]}}
+{% endfor %}
+--
+
+Where:
+
+* `object_name` is the name of the existing context that contains the object
+* `{{item[0]}}` contains the key of the current enumrated object
+* `{{item[1]}}` contains the value
+* `{% endfor %}` indicates where the object enumeration block ends
+
+
+EXAMPLE:
+--
+Given:
+
+object.json
+[source,json]
+----
+---
+name: Lorem ipsum
+desc: dolor sit amet
+----
+
+And the block:
+[source,asciidoc]
+------
+[json2text,object.json,my_item]
+----
+{% for item in my_item %}
+=== {{item[0]}}
+
+{{item[1]}}
+
+{% endfor %}
+----
+------
+
+Where:
+
+* file path is `object.json`
+* current key within the enumerator is called `item[0]`
+* `{{item[0]}}` gives the key name in the current iteration
+* `{{item[1]}}` gives the value in the current iteration
+
+Will render as:
+[source,text]
+----
+=== name
+
+Lorem ipsum
+
+=== desc
+
+dolor sit amet
+----
+
+--
+
+
+
+Moreover, the `keys` and `values` attributes can also be used in object enumerators.
+
+
+EXAMPLE:
+--
+Given:
+
+object.json
+[source,json]
+----
+---
+name: Lorem ipsum
+desc: dolor sit amet
+----
+
+And the block:
+[source,asciidoc]
+------
+[json2text,object.json,item]
+----
+.{{item.values[1]}}
+[%noheader,cols="h,1"]
+|===
+{% for elem in item %}
+| {{elem[0]}} | {{elem[1]}}
+
+{% endfor %}
+|===
+----
+------
+
+Where:
+
+* file path is `object.json`
+* current key within the enumerator is called `key`
+* `{{item[1]}}` gives the value of key in the current iteration the parent array `my_item`.
+* `{{item.values[1]}}` gives the value located at the second key within `item`
+
+Will render as:
+[source,text]
+----
+.dolor sit amet
+
+[%noheader,cols="h,1"]
+|===
+| name | Lorem ipsum
+| desc | dolor sit amet
+|===
+----
+
+--
+
+There are several optional arguments to the for tag that can influence which items you receive in your loop and what order they appear in:
```

Comment:
Delimit the "for" with backticks: "`for`"

Reply:
Changed